File: Java/AA_Lab3.java
import java.util.*;
class Node{
    int data;
    Node next;

    Node(int d){
        this.data=d;
        this.next=null;
    }
}

class Queue{

    Node head;

    public void push(int d){
        Node newnode = new Node(d);
        if (head==null){
            head=newnode;
        }
        else{
            Node temp=head;
            while(temp.next!=null){
                temp=temp.next;
            }
            temp.next=newnode;
        }
    }

    public void pop(){
        if(head==null){
            System.out.println("underflow");
            return;
        }
        System.out.println(head.data+" Item Poped");
        head=head.next;
    }

    public void print(){
        if(head==null){
            System.out.println("underflow ");
            return;
        }
        Node temp=head;
        while(temp!=null){
            System.out.println(temp.data);
            temp=temp.next;
        }
    }

    public void peek(){
        if(head==null){
            System.out.println("underflow");
            return;
        }
        System.out.println(head.data+" Peek Item");
    }
}
public class AA_Lab3{
    public static void main(String[] args) {
        Queue q1=new Queue();
        Scanner in = new Scanner(System.in);
        int d;
        while(true){
            System.out.println("""
                    1.push
                    2.pop
                    3.peek
                    4.print 
                    5.Exit
                    """);
            d=in.nextInt();
            if(d==5){
                break;
            }
            if(d==1){
                System.out.println("Enter Number: ");
                int data=in.nextInt();
                q1.push(data);
            }
            if(d==2){
                q1.pop();
            }
            if(d==3){
                q1.peek();
            }
            if(d==4){
                q1.print();
            }
        }
        in.close();
    }
}

File: AdvancedAlgorithm/AA_Lab3.py
class Node:
    def __init__(self,d):
        self.data=d
        self.next=None
        
class Queue:
    
    def __init__(self):
        self.head=None
        
    def enqueue(self,d):
        Newnode=Node(d)
        if (self.head==None):
            self.head=Newnode
        else:
            temp=self.head
            while(temp.next!=None):
                temp=temp.next
            temp.next=Newnode
    
    def dequeue(self):
        if(self.head==None):
            print("Underflow")
            return
        self.head=self.head.next
        
    def top(self):
        if(self.head==None):
            print("Underflow")
            return
        print(self.head.data)
          
    def display(self):
        if(self.head==None):
            print("Underflow")
            return
        temp=self.head
        while(temp!=None):
            print(temp.data)
            temp=temp.next
            
q1=Queue()
while (True):
    print("""
          Enter Choice-
          1.Push
          2.Pop
          3.Top
          4.Print
          5.Exit\n
          """)
    d=int(input())

    if(d==1):
        print("Enter Data: ")
        data=int(input())
        q1.enqueue(data)
    elif(d==2):
        q1.dequeue()
    elif(d==3):
        q1.top()
    elif(d==4):
        q1.display()
    elif(d==5):
        break
    else:
        print("Enter correct choice")